src/Microsoft.Azure.WebJobs.Host/Executors/JobHostConfigurationExtensions.cs

```diff
@@ -110,6 +110,7 @@ public static ServiceProviderWrapper CreateStaticServices(this JobHostConfigurat
             {
                 loggerFactory = new LoggerFactory();
                 services.AddService<ILoggerFactory>(loggerFactory);
+                config.LoggerFactory = loggerFactory;
```

Comment:
@brettsam - Added loggerfactory to the config object. Is this ok? Other option is to add NullHostInstanceLogger as part of services in [Jobhostconfiguration](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs#L110)

Reply:
Let's remove this entire `if` block, and add the `NullLoggerFactory.Instance` directly to JobHostConfiguration. It makes it easier to reason through how this works.

Reply:
Actually, I take that back. Instead of the `NullLoggerFactory`, use a `new LoggerFactory()`. That way people can immediately start adding to it without needing to new-up another one.